Library/Homebrew/tap.rb

```diff
@@ -280,4 +316,93 @@ def self.each
   def self.names
     map(&:name)
   end
+
+  private
+
+  # @private
+  def formula_file_to_name(file)
+    "#{name}/#{file.basename(".rb")}"
+  end
+
+  # @private
+  def alias_file_to_name(file)
+    "#{name}/#{file.basename}"
+  end
+end
+
+class CoreFormulaTap < Tap
+  # @private
+  def initialize
+    @user = "Homebrew"
+    @repo = "homebrew"
+    @name = "Homebrew/homebrew"
+    @path = HOMEBREW_REPOSITORY
+  end
+
+  def self.instance
+    @instance ||= CoreFormulaTap.new
+  end
+
+  # @private
+  def uninstall
+    raise "Tap#uninstall is not available for CoreFormulaTap"
+  end
+
+  # @private
+  def pin
+    raise "Tap#pin is not available for CoreFormulaTap"
+  end
+
+  # @private
+  def unpin
+    raise "Tap#unpin is not available for CoreFormulaTap"
+  end
+
+  # @private
+  def pinned?
+    false
+  end
+
+  # @private
+  def command_files
+    []
+  end
+
+  # @private
+  def custom_remote?
+    remote != "https://github.com/#{user}/#{repo}.git"
+  end
+
+  # @private
+  def core_formula_tap?
+    true
+  end
+
+  # @private
+  def formula_dir
+    HOMEBREW_LIBRARY/"Formula"
+  end
+
+  # @private
+  def alias_dir
+    HOMEBREW_LIBRARY/"Aliases"
+  end
+
+  # @private
+  def formula_renames
+    require "formula_renames"
```

Comment:
Might be reasonable to change to load if we plan to use it in update, because we can require only once.